server/odc-server/src/main/java/com/oceanbase/odc/server/config/MvcConfiguration.java

```diff
@@ -115,15 +116,28 @@ public void addFormatters(FormatterRegistry registry) {
 
     @Override
     public void extendMessageConverters(List<HttpMessageConverter<?>> converters) {
-        for (HttpMessageConverter converter : converters) {
+        int xmlConverterIndex = -1;
+        int jacksonConverterIndex = -1;
+        for (int i = 0; i < converters.size(); i++) {
+            HttpMessageConverter<?> converter = converters.get(i);
+            if (converter instanceof MappingJackson2XmlHttpMessageConverter) {
+                xmlConverterIndex = i;
+            }
             if (converter instanceof MappingJackson2HttpMessageConverter) {
+                jacksonConverterIndex = i;
                 MappingJackson2HttpMessageConverter jacksonConverter = (MappingJackson2HttpMessageConverter) converter;
                 jacksonConverter.setObjectMapper(objectMapper);
                 log.info("update objectMapper for jackson http message converter");
-                return;
             }
         }
-        throw new RuntimeException("no jackson http message converter found!");
+        if (jacksonConverterIndex == -1) {
+            throw new RuntimeException("no jackson http message converter found!");
+        }
+        if (xmlConverterIndex != -1) {
```

Comment:
May use  `if (xmlConverterIndex != -1 && xmlConverterIndex < jacksonConverterIndex)` in case that the xmlConverter is already after jacksonConverter?

Reply:
nice suggestion! Updated.